sql/core/src/main/scala/org/apache/spark/sql/execution/ui/AllExecutionsPage.scala

```diff
@@ -121,65 +122,247 @@ private[ui] class AllExecutionsPage(parent: SQLTab) extends WebUIPage("") with L
+      s"#$tableHeaderId"
+  }
+
+  override def pageSizeFormField: String = executionTag + ".pageSize"
+
+  override def pageNumberFormField: String = executionTag + ".page"
+
+  override def goButtonFormPath: String = {
+    val encodedSortColumn = URLEncoder.encode(sortColumn, "UTF-8")
+    s"$parameterPath&$executionTag.sort=$encodedSortColumn&$executionTag.desc=$desc#$tableHeaderId"
+  }
+
+  override def headers: Seq[Node] = {
+    // Information for each header: title, cssClass, and sortable
+    val executionHeadersAndCssClasses: Seq[(String, String, Boolean)] =
+      Seq(("ID", "", true), ("Description", "", true), ("Submitted", "", true),
+        ("Duration", "", true)) ++ {
+        if (showRunningJobs && showSucceededJobs && showFailedJobs) {
+          Seq(("Running Job IDs", "", true), ("Succeeded Job IDs", "", true),
+            ("Failed Job IDs", "", true))
+        } else if (showSucceededJobs && showFailedJobs) {
+          Seq(("Succeeded Job IDs", "", true), ("Failed Job IDs", "", true))
+        }
+        else {
+          Seq(("Job IDs", "", true))
+        }
+      }
+
+    if (!executionHeadersAndCssClasses.filter(_._3).map(_._1).contains(sortColumn)) {
+      throw new IllegalArgumentException(s"Unknown column: $sortColumn")
+    }
+
+    val headerRow: Seq[Node] = {
+      executionHeadersAndCssClasses.map { case (header, cssClass, sortable) =>
+        if (header == sortColumn) {
+          val headerLink = Unparsed(
+            parameterPath +
+              s"&$executionTag.sort=${URLEncoder.encode(header, "UTF-8")}" +
+              s"&$executionTag.desc=${!desc}" +
+              s"&$executionTag.pageSize=$pageSize" +
+              s"#$tableHeaderId")
+          val arrow = if (desc) "&#x25BE;" else "&#x25B4;" // UP or DOWN
+
+          <th class={cssClass}>
+            <a href={headerLink}>
+              {header}<span>
+              &nbsp;{Unparsed(arrow)}
+            </span>
+            </a>
+          </th>
         } else {
-          None
+          if (sortable) {
+            val headerLink = Unparsed(
+              parameterPath +
+                s"&$executionTag.sort=${URLEncoder.encode(header, "UTF-8")}" +
+                s"&$executionTag.pageSize=$pageSize" +
+                s"#$tableHeaderId")
+
+            <th class={cssClass}>
+              <a href={headerLink}>
+                {header}
+              </a>
+            </th>
+          } else {
+            <th class={cssClass}>
+              {header}
+            </th>
+          }
         }
+      }
+    }
+    <thead>
+      {headerRow}
+    </thead>
+  }
+
+  override def row(executionTableRow: ExecutionTableRowData): Seq[Node] = {
+    val executionUIData = executionTableRow.executionUIData
+    val submissionTime = executionUIData.submissionTime
+    val duration = executionTableRow.duration
+
+    def jobLinks(status: JobExecutionStatus): Seq[Node] = {
+      executionUIData.jobs.flatMap {
+        case (jobId, jobStatus) =>
```

Comment:
Might be a little more idiomatic to say...

```
case (jobId, jobStatus) if jobStatus == status => ...
case _ => none
```

but why not just filter first instead of flatMap?

Reply:
Thanks. I have used filter and map

Reply:
`.filter { case (_, jobStatus) => jobStatus == status }`? I'm on the fence about whether it's worth it, but I generally like this clearer style. You can also omit `jobStatus` in the case statement for a little more conciseness.